### PR TITLE
Parse support for instrument requests

### PR DESCRIPTION
**Pull request by alastair**
https://github.com/alastair/python-musicbrainzngs/issues/182 and https://github.com/alastair/python-musicbrainzngs/issues/142

Still need tests, and parsing of search results (`instrument-list`).

Documentation http://musicbrainz.org/doc/Development/XML_Web_Service/Version_2#Browse mentions browse instruments by release, but @mwiencek says that this is not implemented and seems to be a bug, so I'm not implementing it.

### CODE REVIEW

File: musicbrainzngs/musicbrainz.py [hunk 1]
```diff
@@ -44,7 +44,7 @@
     ],
     'instrument': [
 
-    ],
+    ] + RELATION_INCLUDES,
```
Comment (alastair) [inline musicbrainzngs/musicbrainz.py]: missing annotation, tag_includes, rating_includes

**Comment by mwiencek** [comment]
I've updated the docs to remove the browse-instruments-by-release reference and opened https://github.com/metabrainz/musicbrainz-server/pull/126 to kill the unimplemented musicbrainz-server code.

**Comment by alastair** [comment]
Should be done now

File: musicbrainzngs/mbxml.py [hunk 2]
```diff
@@ -293,6 +295,19 @@ def parse_event(event):
 
     return result
 
+def parse_instrument(instrument):
+    result = {}
+    attribs = ["id", "type", "ext:score"]
+    elements = ["name", "description"]
```
Comment (mineo) [inline musicbrainzngs/mbxml.py]: According to https://github.com/metabrainz/mmd-schema/blob/master/schema/musicbrainz_mmd-2.0.rng#L912 instruments can have disambiguation comments and https://beta.musicbrainz.org/instrument/dabdeb41-560f-4d84-aa6a-cf22349326fe does indeed have one.

File: test/test_mbxml_instrument.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        musicbrainzngs.set_useragent("test", "1")
+        musicbrainzngs.set_rate_limit(False)
+
+    def testGetInstrument(self):
+        # Tags
+        musicbrainzngs.get_instrument_by_id("")
+
+        # some rels
+
+        # alias, annotation
```
Comment (mineo) [inline test/test_mbxml_instrument.py]: Something seems to be missing here.

Comment (alastair) [inline test/test_mbxml_instrument.py]: whoops, thanks!

Comment (alastair) [inline test/test_mbxml_instrument.py]: fixed

File: musicbrainzngs/mbxml.py (hunk 2 above)
Comment (alastair) [inline musicbrainzngs/mbxml.py]: added

**Comment by alastair** [comment]
(I just force-pushed to add a signed-off-by)

File: test/test_getentity.py [hunk 4]
```diff
@@ -115,3 +115,27 @@ def testGetByDiscid(self):
...
+
+        # alias, annotation
+        musicbrainzngs.get_instrument_by_id("d00cec5f-f9bc-4235-a54f-6639a02d4e4c", includes=["aliases", "annotation"])
+        self.assertEqual("http://musicbrainz.org/ws/2/instrument/d00cec5f-f9bc-4235-a54f-6639a02d4e4c?inc=aliases+annotation", self.opener.get_url())
+
+        # Ratings are used on almost all other entites but instrument
+        try:
+            musicbrainzngs.get_instrument_by_id("dabdeb41-560f-4d84-aa6a-cf22349326fe", includes=["ratings"])
+            self.assertFalse("Should not be able to include ratings")
+        except musicbrainzngs.InvalidIncludeError:
```
Comment (mineo) [inline test/test_getentity.py]: The rest of this file uses `assertRaises` for exceptions.

**Comment by mineo** [comment]
The rest looks good to me.

**Pull request merged**